scripts/vet.sh

```diff
@@ -38,160 +38,172 @@ elif [[ "$#" -ne 0 ]]; then
+
+#   # Usage of the deprecated Logger interface from prefix_logger.go is the only
+#   # allowed one. If any other files use the deprecated interface, this check
+#   # will fails. Also, note that this same deprecation notice is also added to
+#   # the list of ignored notices down below to allow for the usage in
+#   # prefix_logger.go to not case vet failure.
+#   noret_grep "(SA1019)" "${SC_OUT}" | noret_grep "internal.Logger is deprecated:" | not grep -v -e "grpclog/logger.go"
+
+#   # Only ignore the following deprecated types/fields/functions and exclude
+#   # generated code.
+#   noret_grep "(SA1019)" "${SC_OUT}" | not grep -Fv 'XXXXX PleaseIgnoreUnused
+# XXXXX Protobuf related deprecation errors:
+# "github.com/golang/protobuf
+# .pb.go:
+# grpc_testing_not_regenerate
+# : ptypes.
+# proto.RegisterType
+# XXXXX gRPC internal usage deprecation errors:
+# "google.golang.org/grpc
+# : grpc.
+# : v1alpha.
+# : v1alphareflectionpb.
+# BalancerAttributes is deprecated:
+# CredsBundle is deprecated:
+# internal.Logger is deprecated:
+# Metadata is deprecated: use Attributes instead.
+# NewSubConn is deprecated:
+# OverrideServerName is deprecated:
+# RemoveSubConn is deprecated:
+# SecurityVersion is deprecated:
+# Target is deprecated: Use the Target field in the BuildOptions instead.
+# UpdateAddresses is deprecated:
+# UpdateSubConnState is deprecated:
+# balancer.ErrTransientFailure is deprecated:
+# grpc/reflection/v1alpha/reflection.proto
+# SwitchTo is deprecated:
+# XXXXX xDS deprecated fields we support
+# .ExactMatch
+# .PrefixMatch
+# .SafeRegexMatch
+# .SuffixMatch
+# GetContainsMatch
+# GetExactMatch
+# GetMatchSubjectAltNames
+# GetPrefixMatch
+# GetSafeRegexMatch
+# GetSuffixMatch
+# GetTlsCertificateCertificateProviderInstance
+# GetValidationContextCertificateProviderInstance
+# XXXXX PleaseIgnoreUnused'
+#   popd
+# done
+
+# # Error for violation of enabled lint rules in config excluding generated code.
+# revive \
+#   -set_exit_status=1 \
+#   -exclude "reflection/test/grpc_testing_not_regenerate/" \
+#   -exclude "**/*.pb.go" \
+#   -formatter plain \
+#   -config "$(dirname "$0")/revive.toml" \
+#   ./...
+
+# Check dependencies of all non-internal and non-testing packages.
+OUTDIR="$(realpath "$(dirname "$0")")/deps"
+find . -name '*.go' | grep -v '/\(internal\|examples\|benchmark\|interop\|test\|testdata\)/' | xargs dirname | sort -u | while read d; do
+  pushd "$d"
+  pkg="$(echo "$d" | sed 's;^\./;;' | sed 's;\.;grpc;' | sed 's;/;_;g')"
+  go list -deps . | sort >| "${OUTDIR}/$pkg"
   popd
 done
 
-# Error for violation of enabled lint rules in config excluding generated code.
-revive \
-  -set_exit_status=1 \
-  -exclude "reflection/test/grpc_testing_not_regenerate/" \
-  -exclude "**/*.pb.go" \
-  -formatter plain \
-  -config "$(dirname "$0")/revive.toml" \
-  ./...
+git status --porcelain 2>&1 | fail_on_output || \
```

Comment:
Does this mean everytime the deps of any package changes, this check would fail?

Reply:
Yes, basically..  We can make it easy to regenerate the deps if it's not easy enough (basically just copy/paste this loop onto the commandline).